packages/core/nuxt-theme-module/theme/components/AppHeader.vue

```diff
@@ -104,14 +159,66 @@ export default {
       await loadWishlist();
     });
 
+    const handleSearchInput = (input) => term.value = input;
+
+    const closeSearchAndRemoveTerm = () => {
+      term.value = '';
+      searchOpen.value = false;
+      changeSearchTerm(term.value);
+    };
+
+    const handleSearch = async (e) => {
+      const term = e.target.value;
+      changeSearchTerm(term);
+      await search({ categorySlug: 'women-clothing-jackets', itemsPrePage: 20, filters: {} });
+    };
+
+    const debounceSearchInput = debounce(async () => {
+      await search({ categorySlug: 'women-clothing-jackets', itemsPrePage: 20, filters: {} });
+    }, 1000);
+
+    const isMobile = computed(() => mapMobileObserver().isMobile.get());
+
+    const handleCloseSearchButton = () => {
+      if (isMobile.value) {
+        return closeSearchAndRemoveTerm();
+      } else {
+        term.value = '';
+        return searchRef.value.$el.children[0].focus();
+      }
+    };
+
+    watch(() => term.value, (newVal, oldVal) => {
+      if (term.value.length >= 3) {
+        debounceSearchInput();
+      }
+      if (!isMobile.value && term.value.length > 0) {
```

Comment:
make it a single if and maybe assign to a variable
const canDoSth = (!oldVal && newVal) || (newVal.length !== oldVal.length && searchOpen.value === false && ((!isMobile.value && term.value.length > 0))